cocotb/scoreboard.py

```diff
@@ -97,7 +97,9 @@ def compare(self, got, exp, log, strict_type=True):
             self.errors += 1
             log.error("Received transaction is a different type to expected "
                       "transaction")
-            log.info("Got: %s but expected %s" %
+            log.info("Got: %s but expected %s "
+                     "Set strict_type=False when calling add_interface or "
```

Comment:
I don't think we want this part prtined every mismatch as would just clog the output. First time maybe ok.

Reply:
I'd guess that is something the user is fixing very quickly, so clogging the output shouldn't occur for long? But maybe just adding `` (strict_type=False)`` at the end is enough of a hint.

Reply:
So it would only print out once unless the user specified not to fail immediately.  A solution to guarantee it would only print out once (without some flag) would be to put the message in the TestFailure exception when it gets raised.

Reply:
In fact, am I correct in thinking that if we don't fail immediately then strict type checking won't cause a failure at all?  It seems that the compare function will just return, which seems to indicate there was not an issue with the compare. Can someone confirm this?

Reply:
@cmarqu @stuarthodgson Please see 
5480a78